src/Domain.LinnApps/PurchaseOrderReqs/PurchaseOrderReqService.cs

```diff
@@ -110,6 +124,86 @@ public PurchaseOrderReq Create(PurchaseOrderReq entity, IEnumerable<string> priv
             return entity;
         }
 
+        public void CreateOrderFromReq(PurchaseOrderReq entity, IEnumerable<string> privileges, int currentUserId)
+        {
+            if (entity.State != "ORDER WAIT")
+            {
+                throw new UnauthorisedActionException(
+                    "Cannot create order from a req that is not in state 'ORDER WAIT'. Please make sure the req is saved in this state and try again");
+            }
+
+            if (!entity.AuthorisedById.HasValue)
+            {
+                throw new UnauthorisedActionException("Cannot create order from a req that has not been authorised");
+            }
+
+            if (!entity.TotalReqPrice.HasValue)
+            {
+                throw new UnauthorisedActionException("Cannot create order from a req without value for price");
```

Comment:
would be tempted to use a different exception type here

Reply:
Yeah that's probably fair, think a DomainException or a custom "no value" sort of one? A generic-y RequiredButNoValueException or a slightly improved name could be useful elsewhere I'm sure

Reply:
might use ArgumentException